src/webgpu/api/operation/device/lost.spec.ts

```diff
@@ -2,24 +2,82 @@ export const description = `
 Tests for GPUDevice.lost.
 `;
 
-import { Fixture } from '../../../../common/framework/fixture.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { attemptGarbageCollection } from '../../../../common/util/collect_garbage.js';
+import { getGPU } from '../../../../common/util/navigator_gpu.js';
+import { assert, raceWithRejectOnTimeout } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
 
-export const g = makeTestGroup(Fixture);
+class DeviceLostTests extends GPUTest {
+  // Default timeout for waiting for device lost is 2 seconds.
+  readonly kDeviceLostTimeoutMS = 2000;
+
+  getDeviceLostWithTimeout(device: GPUDevice = this.device): Promise<GPUDeviceLostInfo> {
+    return raceWithRejectOnTimeout(
+      this.device.lost,
+      this.kDeviceLostTimeoutMS,
+      'device was not lost'
+    );
+  }
+
+  expectDeviceDestroyed(): void {
+    this.eventualAsyncExpectation(async niceStack => {
+      try {
+        const lost = await this.getDeviceLostWithTimeout();
+        this.expect(lost.reason === 'destroyed', 'device was lost from destroy');
+      } catch (ex) {
+        niceStack.message = 'device was not lost';
+        this.rec.expectationFailed(niceStack);
+      }
+    });
+  }
+}
+
+export const g = makeTestGroup(DeviceLostTests);
 
 g.test('not_lost_on_gc')
   .desc(
     `'lost' is never resolved by GPUDevice being garbage collected (with attemptGarbageCollection).`
   )
-  .unimplemented();
+  .fn(async t => {
+    // Create a new device for this because it is hard to lose all refs to the test fixture device.
+    const adapter = await getGPU().requestAdapter();
+    assert(adapter !== null);
+    let device: GPUDevice | undefined = await adapter.requestDevice();
+    assert(device !== null);
+
+    t.shouldReject('Error', t.getDeviceLostWithTimeout(device), 'device was unexpectedly lost');
+
+    device = undefined;
+    await attemptGarbageCollection();
```

Comment:
Tracking `device` to make sure it actually goes out of scope is a little difficult, since it gets passed getDeviceLostWithTimeout - and the escape analysis of JS VMs might have the same problem (and keep it alive even though it doesn't need to). To try to make as much stuff go directly go out of scope, how about tightly constraining them to a function scope something like:

```ts
const lost = await (async () => {
  const adapter = await getGPU().requestAdapter();
  assert(adapter !== null);
  return (await adapter.requestDevice()).lost;
})();
```

It's inherently impossible to guarantee things get garbage collected in JS but I think this at least makes it more likely.

It would be interesting to try to detect GC using WeakRef - this could just be a TODO for now if you don't want to get into it. We could have another copy of this test (that skips itself if WeakRef isn't available) that uses WeakRef to detect if the GPUDevice got GC'd, and produces a warning if it didn't. Right now warnings don't count as failure, but eventually I think we will start running a warning-as-error mode in Chromium.

Reply:
```ts
const { lost } = await (async () => {
  const adapter = await getGPU().requestAdapter();
  assert(adapter !== null);
  const lost = (await adapter.requestDevice()).lost;
  return { lost };
})();
```

Reply:
Done